### DISCUSSION TITLE
Invoice document for small business without VAT + different templates for timesheet pdfs?

### DISCUSSION THREAD

**Discussion by LiaraAlis** [kimai/kimai#3478] [Questions & Answers] [answered]
Hello together,

I searched a long time for a tool where can I book my times for my small business and my mini job. I've found Kimai now and I really like it.
But I have two questions:
1. I run a small business in germany which is exempt from the VAT. So I need a invoice template that has no VAT on it. In the template settings I've added a note about this in the "payment settings" field. But I need also a possibility to remove the VAT on the pdf. Is the only way to edit the pdf template?
2. For my mini job I only need a export for entered times. Is there a way to remove some columns in the time sheet pdf for a specific customer? In the print view I can select which fields are output. Is this also possible for the pdf? But I would like to keep the standard for other customers.

I can't find any information about these two questions in the issues or discussions. So I hope you can help me.

Thanks in advance! :)

**Comment by LiaraAlis** [comment, marked as answer]
Okay, I found a solution for the second question. I just created a new export template and added a translation key for this as described in the documentation:
https://www.kimai.org/documentation/export.html#adding-export-templates

**Reply by kevinpapst** [reply]
You can set tax to 0 (what I do for certain invoices).
Additionally you can edit the invoice template and remove this block:
https://github.com/kevinpapst/kimai2/blob/master/templates/invoice/renderer/default-pdf.pdf.twig#L139-L144

**Reply by LiaraAlis** [reply]
Thanks for your answer. I know that, but was hoping there might be an easier solution.
And as I see there are no twig blocks in this template. Is there a reason for that?
As I said I edited the export pdf template before and there are blocks that I was able to override. Why not in the invoice template?

**Reply by kevinpapst** [reply]
Simpler than setting tax rate to 0?

Regarding the why: sent a PR and we'll see if we have blocks in the future.

**Reply by LiaraAlis** [reply]
Setting the tax rate to 0 doesn't hide the tax from the invoice. Only 0% is shown in the invoice.
In my opinion it is better to hide the tax completely in my case and add the note to the invoice.
If it's okay, I will make a PR to add this feature.

**Reply by kevinpapst** [reply]
> In my opinion it is better to hide the tax completely

Multiple tax consultants had a look at it and dozens of companies received such invoices (with 0% tax) - nobody ever cared.
Sounds like the idea that invoice numbers need to be consecutive 😁 

But sure, please sent a PR.

**Reply by LiaraAlis** [reply]
Haha, I'm not a tax consultant or something else. I only know it this way from my previous invoicing program and other programs I tried (like Papierkram.de). :D
Maybe it is not needed to do this. But I like it more. xD

I'll look at this. ^^

**Reply by kevinpapst** [reply]
Yeah true, better safe than sorry!

Maybe we should change the template directly and add a check for zero. 
If tax rate = zero then hide tax rate row.
I don't think there is any reason to include the row with a zero rate. I just ignored this issue forever.

**Reply by LiaraAlis** [reply]
No problem, I have an idea to solve this. I will make a pr and maybe it's OK for you. :)

**Reply by LiaraAlis** [reply]
@kevinpapst  Here you can find the PR: https://github.com/kevinpapst/kimai2/pull/3480 😁

**Reply by LiaraAlis** [reply]
Here you can find the first dev version of my plugin: https://github.com/LiaraAlis/kimai2-SmallBusinessRuleBundle

**Reply by kevinpapst** [reply]
Wow, you seem to know a bit about Symfony. 

Some minor feedback:

- This looks like a copy & paste mistake: https://github.com/LiaraAlis/kimai2-SmallBusinessRuleBundle/blob/60d191d5e98397cf68380f3cf6c4403c59177247/EventSubscriber/PluginActionsSubscriber.php#L62
- Instead of using the form type, can't you use a FormExtension instead? Probably not necessary, just feels more natural to me.
- This line can be removed: https://github.com/LiaraAlis/kimai2-SmallBusinessRuleBundle/blob/master/composer.json#L21

I really like the solution, to put that entire "small business" logic into a bundle. 
It is impressive to see what can be achieved with Symfony bundles, if the dev knows the framework.

If you are ever looking for a new job, give me a ping 😄

**Reply by LiaraAlis** [reply]
Thanks for the tips and your feedback. :)
I will have a look about the form extension. Maybe it is a more elegant way to solve this.

I'm not an expert for symfony. But I currently work primarily with Shopware 6 which is a symfony application. So I have already seen some things and also faced one or the other problem. 😁

**Reply by LiaraAlis** [reply]
I've updated the v1.0-dev version. If there is nothing more, I would like to add it to the marketplace. I will push the changes to the website repository in the next days. 😁

**Reply by kevinpapst** [reply]
Awesome, looking forward to have a new plugin I the marketplace 😄

**Comment by LiaraAlis** [comment]
@kevinpapst First, thank you very much for your support solving this topic.

At the moment, I am creating a invoice template to match my requirements. I noticed some discrepancies while doing this:
1. The translation label for the own VAT ID and the VAT ID of the customer is the same. So it's not clear which is which when printing both on the invoice. I think we should split these two meanings.
2. Small business doesn't need a VAT ID if they use the small business regulation, but they need to print their tax number on the invoice. Kimai has no field for that. Of course, we can use the VAT ID field for that. But due to the problem pointed in 1 I cannot change the label in the translation to print "Tax number" instead of "VAT ID" and this could be a problem if you want to print also the VAT ID of the customer (why ever).
I thinked about adding this field within my plugin and adding the correct translation for this. Or to replace the label for the VAT ID field in the form. But this doesn't solve the problem in the invoice template.

What do you think?

**Reply by kevinpapst** [reply]
1. How and why? You print the VAT of you customer at the customer address and your VAT in your address section. That is how everyone I met does it. So its always clear which VAT belongs to whom.
2. You are thinking like a developer 😁 As a user I would probably try to add a new line to my own address "Tax number: xxxxxxx" and leave the VAT empty (so it does not appear in the PDF). Problem solved hihihihi

But as you know I am totally open to hear your improvement ideas!

**Reply by LiaraAlis** [reply]
1. I think not. I've never seen that the custom VAT ID is printed at the customer address. I know it rather that this is listed in the block to the right of the recipient's address. I therefore find it better to separate the texts.
2. Of course, this is possible. I just thought this today on the way to work. 😆 But it is more flexible if there is a separate field for this. Would it be bad to introduce such a field into the standard? I can do that in the plugin but I think it fits more to the standard.

**Reply by LiaraAlis** [reply]
Additionally, I would suggest to add a field in invoice option to enter a introduction text like "Thank you for your order. I hereby invoice you for the following work."

**Reply by kevinpapst** [reply]
Well, the core system cannot satisfy every possible use-case. Instead it should follow the 80:20 rule and as nobody ever asked for it before, I would recommend to add it to your plugin for now. I don't quite understand why you need a new field. The field is there for the tax number, no matter if you call it Umsatztsteuer ID, VAT no or SteuerID. 
I understand that this makes it a bit more complicated for you, as it is not automatically available to every template out there. But maybe you can ship your ideal small-business template with your bundle, including the translation and the the additional configurations.

Additionally, these small business regulations are likely only available in a handful of countries. There are many countries who could benefit from a second tax rate, which I currently do not plan to add as well. And so many other things that a full fledged invoicing software might have to support, but at the end invoicing 

Anyway: I don't mind adding two translations, something along the line of "Your/Customer/Recipient VAT number" and "Our VAT number"? But we have to keep an eye on BC, so we cannot simply change the existing label and all invoice templates out there, which rely on the current translation.

**Reply by LiaraAlis** [reply]
I understand your point.
Is there a way to change translations update-safe? So editing the language files manually is not very elegant.

I also thinked about publishing my template for the public. I'm trying to create it to match the requirements mentioned in DIN 5008.
But I'm thinking about creating a separate repository where we can collect different templates. I saw that this has existed in the past and it is still linked in the docs (https://github.com/Keleo/kimai2-invoice-templates). So I can link to this repository in the README of my plugin.

> And so many other things that a full fledged invoicing software might have to support, but at the end invoicing
But and… what? 😋

> Anyway: I don't mind adding two translations, something along the line of "Your/Customer/Recipient VAT number" and "Our VAT number"? But we have to keep an eye on BC, so we cannot simply change the existing label and all invoice templates out there, which rely on the current translation.

At the moment I think I have a solution, so it is not necessary anymore. I will create a new text snippet for the "Steuer ID" in my plugin to use it with the tax number field and use the existing text for the VAT ID of the customer.
But if we do this in the future, I suggest to leave to current snippet key as it is and create two new.
